cv19ResSupportV3/V3/Gateways/ResidentGateway.cs

```diff
@@ -164,37 +165,55 @@ public Resident PatchResident(int id, PatchResident command)
             return _helpRequestsContext.ResidentEntities.Find(id).ToDomain();
         }
 
-
         public int? FindResident(FindResident command)
         {
             try
             {
-                if (command.Uprn != null)
+                if (Predicates.IsNotNullAndNotEmpty(command.NhsNumber))
                 {
-                    var response = _helpRequestsContext.ResidentEntities.FirstOrDefault(x =>
-                        x.Uprn.ToUpper().Trim() == command.Uprn.ToUpper().Trim() &&
-                        x.FirstName.ToUpper().Trim() ==
-                        command.FirstName.ToUpper().Trim() &&
-                        x.LastName.ToUpper()
-                            .Trim() ==
-                        command.LastName.ToUpper().Trim());
-                    if (response != null)
-                    {
-                        return response.Id;
-                    }
+                    var matchingResident = _helpRequestsContext.ResidentEntities
+                        .FirstOrDefault(r => r.NhsNumber.Trim().ToUpper() == command.NhsNumber.Trim().ToUpper());
```

Comment:
Can NhsNumber contain letters? Not sure if we need ```ToUpper``` here

Reply:
Yeah, at the time of writing the code I didn't know that myself. Only figured that out after resolving a merge conflict and forgot to remove it 😁 Will do now.

Reply:
Done.